modules/compute/vm-instance/variables.tf

```diff
@@ -409,3 +409,22 @@ variable "allow_automatic_updates" {
   default     = true
   nullable    = false
 }
+
+variable "reservation_name" {
+  description = <<-EOD
+    Name of the reservation to use for VM resources, should be in one of the following formats:
+    - projects/PROJECT_ID/reservations/RESERVATION_NAME[/SUFF/IX]
+    - RESERVATION_NAME[/SUFF/IX]
+
+    Must be a "SPECIFIC" reservation
+    Set to empty string if using no reservation or automatically-consumed reservations
+  EOD
+  type        = string
+  default     = ""
```

Comment:
Why not make it nullable to distinguish between "unset" and "empty string"?

Reply:
Because then the logic of determining if a user used a null or "" to indicate no reservation becomes more complicated.

Reply:
I don't think it significantly changes the complexity, you only need to wrap `var.reservation_name` in a `coalesce()` call:
```
length(regexall("^((projects/([a-z0-9-]+)/reservations/)?([a-z0-9-]+))?$", coalesce(var.reservation_name, ""))) > 0
```

It may be useful in the future to know with a reasonable degree of confidence whether or not the user set a value or left it unset.  I say "reasonable degree of confidence", because it's always possible for users to explicitly set it to `null`, but that seems generally unlikely to happen.

Reply:
This is to your comment below: Why give the user more options here?  Null and empty-string will end up evaluating the same way, so you're just adding unnecessary complexity if you allow for null.

Additionally this is how it's handled in the slurm-gcp modules (where I took the code)

Reply:
Ahh if it's already standardized elsewhere, that makes sense.  Thanks for the explanation!